assets/dev/js/admin/hints/media.js

```diff
@@ -0,0 +1,110 @@
+				</svg>
+			</div>
+			<div class="elementor-control-notice-main">
+				<div class="elementor-control-notice-main-content"><%= content %></div>
+				<div class="elementor-control-notice-main-actions">
+					<button class="e-btn e-<%= type %> e-btn-1" data-event="<%= button_event %>" data-settings="<%= button_data %>">
+						<%= button_text %>
+					</button>
+				</div>
+			</div>
+			<button class="elementor-control-notice-dismiss" data-event="<%= dismissible %>">
+				<i class="eicon eicon-close" aria-hidden="true"></i>
+				<span class="elementor-screen-only"><%= dismiss %></span>
+			</button>
+		</div>
+	</div>`,
+		template( view ) {
+		// Get the template and parse it
+			const html = wp.media.template( 'attachment-details' )( view ); // The template to extend
+			const dom = document.createElement( 'div' );
+			dom.innerHTML = html;
+
+			if ( ! this.shouldDisplayHint( view ) ) {
+				return dom.innerHTML;
+			}
+
+			const hint = document.createElement( 'div' ); // Create a new element
+			hint.classList.add( 'e-hint' ); // Add a class to the element for styling
+			hint.innerHTML = _.template( this._tmpl )( elementorAdminHints.mediaHint ); // Add the content to the new element
+
+			// Insert the new element at the correct spot
+			const details = dom.querySelector( '.attachment-info' );
+			details.appendChild( hint ); // Add new element at the correct spot
+
+			return dom.innerHTML;
+		},
+
+		events: {
+			...wp.media.view.Attachment.Details.prototype.events,
+			'click .elementor-control-notice-dismiss': 'dismiss',
+			'click .e-hint__container .e-btn-1': 'onHintAction',
+		},
+
+		shouldDisplayHint( view ) {
+			if ( ! elementorAdminHints || ! elementorAdminHints?.mediaHint ) {
+				return false;
+			}
+			if ( window.elementorHints !== undefined ) {
+				return false;
+			}
+			if ( view.type !== 'image' ) {
+				return false;
+			}
+			if ( elementorAdminHints.mediaHint.display ) {
+				return true;
+			}
+			return this.imageNotOptimized( view );
+		},
+
+		imageNotOptimized( attachment ) {
+			const checks = {
+				height: 1080,
+				width: 1920,
+				filesizeInBytes: 100000,
+			};
+
+			return Object.keys( checks ).some( ( key ) => {
+				const value = attachment[ key ] || false;
+				return value && value > checks[ key ];
+			} );
+		},
+
+		onHintAction( event ) {
+			const b64Settings = event.target.closest( 'button' ).dataset.settings;
+			const settings = atob( b64Settings );
+			const { action_url: actionURL = null } = JSON.parse( settings );
+			if ( actionURL ) {
+				window.open( actionURL, '_blank' );
+			}
+			this.dismiss( event );
+		},
+
+		dismiss( event ) {
+			elementorCommon.ajax.addRequest( 'dismissed_editor_notices', {
```

Comment:
Not good, you are using the 'editor notices' mechanism to save data for the WordPress admin area.

Editor notices saves data with "widget name + control name".

See https://github.com/elementor/elementor/blob/main/includes/user.php, I believe there is a separate method for admin nitoces.

Reply:
Nope, you are wrong, this is using the part of saving and retrieving the data but the same way the Notice control does. meaning the dismiss key is hardcoded and has no need for widget or control.

its already used for other hints in this same way.